Makefile

```diff
@@ -1,6 +1,6 @@
 .PHONY: test test-report test-fix update-compatibility-patch
 
-PHP_74_OR_NEWER=`php -r "echo (int) version_compare(PHP_VERSION, '7.4', '>=');"`
+PHP_74_OR_NEWER=$(shell php -r "echo (int) version_compare(PHP_VERSION, '7.4', '>=');")
```

Comment:
why this change?

Reply:
Without this change, the command is run every time the variable is evaluated.

Reply:
Interesting, is that Makefile specific thing? Because this shouldn't be the case in bash. And is same justification used for both changes: changing `` into `$()` AND adding `shell`? Where can I find something more about this?

Reply:
Yes, [it's very much Makefile specific](https://www.gnu.org/software/make/manual/html_node/Flavors.html#Flavors). It looks like an alternative would be to use `:=` note that [using backticks is deprecated in favor of `$()` in bash](https://unix.stackexchange.com/questions/126927/have-backticks-i-e-cmd-in-sh-shells-been-deprecated).
Also note that `$(shell foo)` is a function call and does not result in make calling an hypothetical `shell` binary. I think I am going to go with `:=`, that should be less complicated.

Reply:
Ok I'm going to go with both: https://stackoverflow.com/questions/1435861/computing-makefile-variable-on-assignment